src/backend/plugins/config-kde/config-kde.c

```diff
@@ -0,0 +1,306 @@
+  KDE_PROXY_TYPE_PAC,
+  KDE_PROXY_TYPE_WPAD,
+  KDE_PROXY_TYPE_SYSTEM,
+} KdeProxyType;
+
+struct _PxConfigKde {
+  GObject parent_instance;
+
+  char *config_file;
+  gboolean available;
+  GFileMonitor *monitor;
+
+  GStrv no_proxy;
+  char *http_proxy;
+  char *https_proxy;
+  char *ftp_proxy;
+  char *socks_proxy;
+  KdeProxyType proxy_type;
+  char *pac_script;
+};
+
+G_DEFINE_FINAL_TYPE_WITH_CODE (PxConfigKde,
+                               px_config_kde,
+                               G_TYPE_OBJECT,
+                               G_IMPLEMENT_INTERFACE (PX_TYPE_CONFIG, px_config_iface_init))
+
+enum {
+  PROP_0,
+  PROP_CONFIG_OPTION
+};
+
+static void px_config_kde_set_config_file (PxConfigKde *self,
+                                           char        *proxy_file);
+
+static void
+on_file_changed (GFileMonitor      *monitor,
+                 GFile             *file,
+                 GFile             *other_file,
+                 GFileMonitorEvent  event_type,
+                 gpointer           user_data)
+{
+  PxConfigKde *self = PX_CONFIG_KDE (user_data);
+
+  g_debug ("%s: Reloading configuration\n", __FUNCTION__);
+  px_config_kde_set_config_file (self, g_file_get_path (file));
+}
+
+static void
+px_config_kde_set_config_file (PxConfigKde *self,
+                               char        *proxy_file)
+{
+  g_autoptr (GError) error = NULL;
+  g_autofree char *line = NULL;
+  g_autoptr (GFile) file = NULL;
+  g_autoptr (GFileInputStream) istr = NULL;
+  g_autoptr (GDataInputStream) dstr = NULL;
+  const char *desktops;
+
+  self->available = FALSE;
+
+  desktops = getenv ("XDG_CURRENT_DESKTOP");
+  if (!desktops)
+    return;
+
+  /* Remember that XDG_CURRENT_DESKTOP is a list of strings. */
+  if (strstr (desktops, "KDE") == NULL)
+    return;
+
+  g_clear_pointer (&self->config_file, g_free);
+  self->config_file = proxy_file ? g_strdup (proxy_file) : g_build_filename (g_get_user_config_dir (), "kioslaverc", NULL);
+
+  file = g_file_new_for_path (self->config_file);
+  if (!file) {
+    g_debug ("%s: Could not create file for %s", __FUNCTION__, self->config_file);
+    return;
+  }
+
+  istr = g_file_read (file, NULL, NULL);
+  if (!istr) {
+    g_debug ("%s: Could not read file %s", __FUNCTION__, self->config_file);
+    return;
+  }
+
+  dstr = g_data_input_stream_new (G_INPUT_STREAM (istr));
+  if (!dstr)
+    return;
+
+  g_clear_object (&self->monitor);
+  self->monitor = g_file_monitor (file, G_FILE_MONITOR_NONE, NULL, &error);
+  if (!self->monitor)
+    g_warning ("Could not add a file monitor for %s, error: %s", g_file_get_uri (file), error->message);
+  else
+    g_signal_connect_object (G_OBJECT (self->monitor), "changed", G_CALLBACK (on_file_changed), self, 0);
+
+  do {
+    g_clear_pointer (&line, g_free);
+
+    line = g_data_input_stream_read_line (dstr, NULL, NULL, &error);
+    if (line) {
+      g_auto (GStrv) kv = NULL;
+      g_autoptr (GString) value = NULL;
+      kv = g_strsplit (line, "=", -1);
```

Comment:
With this simple change, `=` in values would be supported:
```suggestion
      kv = g_strsplit (line, "=", 2);
```

Reply:
Changed it.